fs_storage/models/fs_storage.py

```diff
@@ -363,6 +376,12 @@ def _get_filesystem(self) -> fsspec.AbstractFileSystem:
         directory_path = self.directory_path
         if directory_path:
             fs = fsspec.filesystem("rooted_dir", path=directory_path, fs=fs)
+        if self.invalidate_cache_on_error:
+            # Check whether we need to invalidate cache or not
+            try:
+                fs.exists("/")
+            except Exception:
+                fs.clear_instance_cache()
```

Comment:
If it fails should we raise the error??? IMO we should.

Reply:
@lmignon changed it